src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -237,3 +272,36 @@ impl DeviceManager {
         Err(anyhow!("ID are exhausted"))
     }
 }
+
+// Many scenarios have similar steps when adding devices. so to reduce duplicated code,
+// we should create a common method abstracted and use it in various scenarios.
+// do_handle_device:
+// (1) new_device with DeviceConfig and return device_id;
+// (2) try_add_device with device_id and do really add device;
+// (3) return device info of device's info;
+pub async fn do_handle_device(
+    d: &RwLock<DeviceManager>,
```

Comment:
Maybe just use &DeviceManager here is better? Let the caller to get RwLock guard.

Reply:
Thx, 
IMO, if so, many dup code will be seen.  I will keep it.

Reply:
OK